### DISCUSSION TITLE
Building NVDA

### DISCUSSION THREAD

**Discussion by kyrylo-pshenychnyi** [nvaccess/nvda#14506] [Q&A] [answered]
Hi everyone, 

I want to build NVDA project for debugging purposes and have an issue while building the project. 

Running the `scons source` command produces the following error: 

![image](https://user-images.githubusercontent.com/45938288/210448105-5d9f6a6a-b009-49ec-9eaa-0cd04c166b98.png)

The C2338 error is connected with a `static_assert` failure. I opened `Microsoft.UiAutomation.h` file and saw that it expects a certain WinRT version (2.200609.3). 

![image](https://user-images.githubusercontent.com/45938288/210448922-acc0b97b-702d-4d98-a37c-e924e2dd3966.png)

As far as I understand this package comes with MSVC build tools - I have the latest version installed. 

![image](https://user-images.githubusercontent.com/45938288/210448666-1b19fcb1-1076-4536-bcc6-69f1d8e73bf4.png)

Could you please help me with this? Any help appreciated.

**Comment by seanbudd** [comment]
Can you confirm the list of VS components you have installed? Are you using VS 2019?

You should have:
* Desktop development with C++
* MSVC v142 - VS 2019 C++ x64/x86 build tools
* Windows 11 SDK (10.0.22000.0)
* C++ ATL for v142 build tools (x86 & x64)
* C++ Clang tools for Windows
* MSVC v142 - VS 2019 C++ ARM64 build tools
* C++ ATL for v142 build tools (ARM64)

**Reply by kyrylo-pshenychnyi** [reply]
Yes, I am using VS 2019. I have all this components installed. Attaching the screenshots of components installed. 
![image](https://user-images.githubusercontent.com/45938288/210528323-d9372597-2364-46f9-b070-d1b37fbacc6a.png)
![image](https://user-images.githubusercontent.com/45938288/210528352-1827c4d2-4d4a-40bc-b408-6fb695b70473.png)
![image](https://user-images.githubusercontent.com/45938288/210528389-4769608a-6c1f-429e-8f83-3a2ed91e3258.png)

**Comment by kyrylo-pshenychnyi** [comment]
It's seems to me I managed to resolve this issue. Ii was connected with the Microsoft SDK version. Windows 11 SDK (10.0.22000.0) must be used. 

However, I face another issue. After the build was finished ('scans source' finished successfully) I run `runnvda.bat` and nothing happens. Am I missing something? 


![image](https://user-images.githubusercontent.com/45938288/210640659-f36c4c9f-ae15-4833-8943-0a8c1a0ec987.png)

**Reply by seanbudd** [reply]
What is the results of running `git status`?

Before running `scons source`, have you done the following?

```py
git submodule sync --recursive
git submodule update --init --recursive
scons source
runnvda.bat
```

**Reply by kyrylo-pshenychnyi** [reply]
I did this steps, but the result is still the same - `runvda.bat` does nothing. I also noticed some strange errors during the build like this:

`Can't read dictionary file: 'C:\Projects\nvda\include\espeak/espeak-ng-data\jbo_dict`. I also attach a video of these steps performed and the output. Hope it's helpful. 


https://user-images.githubusercontent.com/45938288/210668946-372a1a3e-8017-43c1-8572-51c9cdb71074.mp4

**Reply by seanbudd** [reply]
What is the results of running git status? Is the directory clean?

**Reply by seanbudd** [reply]
Performing a `git clean -xdf` than starting again may help. This will clean up all NVDAs build files. Note, this will delete any work you have done on the repository.

**Reply by kyrylo-pshenychnyi** [reply]
My working tree is clear. I tried `git clean -xdf` and result is still the same. I am attaching the `scons source` command output. Hope that helps. Thank you so far for the assistance. 
[output.txt](https://github.com/nvaccess/nvda/files/10351008/output.txt)

**Reply by kyrylo-pshenychnyi** [reply]
I managed to fix the issue. I updated Python (to 3.7.9) and everything worked properly

**Comment by kyrylo-pshenychnyi** [comment, marked as answer]
I managed to fix the issue. I updated Python (to 3.7.9) and everything worked properly
